marshal.go

```diff
@@ -266,6 +266,28 @@ func marshalImpl(vo reflect.Value, w io.Writer, pos uint64, parent *Element, opt
 					}
 					pos, err = writeOne(lst[0])
 				}
+			case reflect.Func:
+				ret := vn.Call(nil)
+				val := ret[0]
```

Comment:
It may be better to check `len(ret) == 1 || len(ret) == 2` before accessing `ret[0]`.

Reply:
Make sense, added the check by 0f4b1af